pyOCD/flash/flash.py

```diff
@@ -70,16 +70,22 @@ class Flash(object):
     def __init__(self, target, flash_algo):
         self.target = target
         self.flash_algo = flash_algo
-        self.end_flash_algo = flash_algo['load_address'] + len(flash_algo)*4
-        self.begin_stack = flash_algo['begin_stack']
-        self.begin_data = flash_algo['begin_data']
-        self.static_base = flash_algo['static_base']
-        self.page_size = flash_algo['page_size']
+        if flash_algo is not None:
+            self.end_flash_algo = flash_algo['load_address'] + len(flash_algo)*4
+            self.begin_stack = flash_algo['begin_stack']
+            self.begin_data = flash_algo['begin_data']
+            self.static_base = flash_algo['static_base']
+            self.page_size = flash_algo['page_size']
 
     def init(self):
         """
         Download the flash algorithm in RAM
         """
+
+        if self.flash_algo is None:
```

Comment:
Rather than in the init, I could put it in flashBinary, flashBlock and getFlashBuilder since those are the primary interface functions for programming.  The only downside would be calling the other functions directly would result in an exception rather than an additional error message.  Let me know what you think.